sdk/typescript/src/types/transactions.ts

```diff
@@ -344,54 +361,54 @@ export function getTransactionGasBudget(tx: SuiTransactionResponse): number {
 export function getTransferObjectTransaction(
   data: SuiTransactionKind,
 ): TransferObject | undefined {
-  return 'TransferObject' in data ? data.TransferObject : undefined;
+  return data.type === 'TransferObject' ? (data as TransferObject) : undefined;
 }
 
 export function getPublishTransaction(
   data: SuiTransactionKind,
 ): SuiMovePackage | undefined {
-  return 'Publish' in data ? data.Publish : undefined;
+  return data.type === 'Publish' ? (data as SuiMovePackage) : undefined;
```

Comment:
Do you need these `as`? I imagine the `type` should narrow these ideally, if they're not then we should figure out what's going wrong with the struct definition such that it isn't narrowing automatically with the type here.